mindmeld/components/entity_resolver.py

```diff
@@ -536,17 +708,67 @@ def _construct_whitelist_query(entity, weight=1, use_phons=False):
 
                 results.append(result)
 
-            return results[0:20]
+            return results
+
+    def _load(self):
```

Comment:
So I am thinking we probably should not do `fit()` in the implementation of `_load()` for all ER models here. I know this is from initial implementation and I think it's another thing that we should re-visit.

My main reasoning is that it would be a more desirable/predictable behavior and it would be more consistent with other components in the NLP pipeline (e.g. domain/intent classifier and entity recognizer) The implementation of `load()` should load whatever model that is trained and serialized by `fit()` and populate the `ready` flag accordingly. This would mean checking on the index existence for Elasticsearch-based model (but not creating it if it does not exist) and loading serialized embeddings for BERT-based model. This way we won't surprise developers when they just want to load already trained model for doing predictions. For example, when they use CLI `converse` command to load application for interactive session. What do you think?

Reply:
I agree with your comment. Like we discussed offline, to keep the changes tractable, let me do this feature as a separate PR.

Reply:
@murali1996 Could you add a TODO in this PR on what will be done in the next PR regarding the fit op?